### PR TITLE
Roe-2413 bo other remove with url params

### PR DESCRIPTION
**Pull request by markpit**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-2413

### Change description

Add url params into the remove button for BO Other page

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Review by mwestacott — APPROVED** [review]

**Review by mattch1 — APPROVED** [review]

File: src/utils/beneficial.owner.other.ts [hunk 1]
```diff
@@ -60,6 +63,13 @@ export const getBeneficialOwnerOtherById = (req: Request, res: Response, next: N
       [StartDateKey]: startDate
     };
 
+    // Redis removal work - Add extra template options if Redis Remove flag is true
+    const isRegistration: boolean = req.path.startsWith(config.LANDING_URL);
+    if (isActiveFeature(config.FEATURE_FLAG_ENABLE_REDIS_REMOVAL) && isRegistration) {
+      templateOptions.FEATURE_FLAG_ENABLE_REDIS_REMOVAL = true;
```
Comment (dlloyd1-ch) [inline src/utils/beneficial.owner.other.ts:69]: Could this be extracted to a method for consistency?  See addCeasedDateToTemplateOptions(). It also uses different syntax to add the filelds.

File: src/utils/beneficial.owner.other.ts [hunk 2]
```diff
@@ -60,6 +63,13 @@ export const getBeneficialOwnerOtherById = (req: Request, res: Response, next: N
       [StartDateKey]: startDate
     };
 
+    // Redis removal work - Add extra template options if Redis Remove flag is true
```
[the remark this answers is not in the document]
Comment (mwestacott) [reply src/utils/beneficial.owner.other.ts:66]: Discussed. Makes it easier later on when we come to do update if this remains where it is.

File: src/utils/beneficial.owner.other.ts (hunk 1 above)
Comment (markpit) [reply src/utils/beneficial.owner.other.ts]: yes can do, i've now extracted it into a new module/file utils/template.data.ts, a function called addActiveSubmissionBasePathToTemplateData.

File: src/utils/template.data.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { isActiveFeature } from "./feature.flag";
+import * as config from "../config";
+import { getUrlWithParamsToPath } from "./url";
+import { Request } from "express";
+
+export const addActiveSubmissionBasePathToTemplateData = (templateData: Object, req: Request) => {
+  if (isActiveFeature(config.FEATURE_FLAG_ENABLE_REDIS_REMOVAL)) {
+    templateData["FEATURE_FLAG_ENABLE_REDIS_REMOVAL"] = true;
```
Comment (mwestacott) [inline src/utils/template.data.ts:8]: I preferred the old way of setting the field data - less verbose, without []s and ""s. Is there an advantage to doing it this way?

File: test/utils/template.data.spec.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      const testObject = {
+        name: "Bob",
+        isOnRegister: false
+      };
+      const request = {} as Request;
+      addActiveSubmissionBasePathToTemplateData(testObject, request);
+
+      expect(testObject["FEATURE_FLAG_ENABLE_REDIS_REMOVAL"]).toEqual(true);
+      expect(testObject["activeSubmissionBasePath"]).toEqual(dummyUrl);
+    });
```
Comment (mwestacott) [inline test/utils/template.data.spec.ts:33]: I'd also check that the existing fields set on the `testObject` are still there and unchanged. Here and in test below.

File: src/utils/template.data.ts (hunk 3 above)
Comment (markpit) [reply src/utils/template.data.ts:8]: not sure, it might be safer with undefined's but i can ind out

File: test/utils/template.data.spec.ts (hunk 4 above)
Comment (markpit) [reply test/utils/template.data.spec.ts:33]: ok, i'll update the test

File: src/utils/template.data.ts (hunk 3 above)
Comment (mwestacott) [reply src/utils/template.data.ts:8]: 👍 if we can avoid the extra characters would be good.

Comment (markpit) [reply src/utils/template.data.ts:8]: can't use the dot as it says property undefined on type Object, so the type has to have the field names in it. Not sure why it seemed to be ok before though.

File: test/utils/template.data.spec.ts (hunk 4 above)
Comment (markpit) [reply test/utils/template.data.spec.ts:35]: updated

File: src/utils/template.data.ts (hunk 3 above)
Comment (mwestacott) [reply src/utils/template.data.ts:8]: Oh well.

**Review by mwestacott — APPROVED** [review]

**Pull request merged**